libsolidity/analysis/TypeChecker.cpp

```diff
@@ -737,6 +737,48 @@ void TypeChecker::endVisit(ExpressionStatement const& _statement)
 			typeError(_statement.expression().location(), "Invalid integer constant.");
 }
 
+bool TypeChecker::visit(Conditional const& _conditional)
+{
+	expectType(_conditional.condition(), BoolType());
+
+	_conditional.trueExpression().accept(*this);
+	_conditional.falseExpression().accept(*this);
+
+	TypePointer const& trueType = type(_conditional.trueExpression());
+	TypePointer const& falseType = type(_conditional.falseExpression());
+
+	TypePointer commonType;
+	if (*trueType == *falseType)
```

Comment:
You can now directly use `TypePointer commonType == Type::commonType(trueType, falseType);`.